### PR TITLE
feat: Support for viewing documentation in-engine

### PR DESCRIPTION
**Pull request by dragmine149**
Changed comments to allow for viewing the documentation in godot on all main functions at least.
Functions beginning with `_` comments were kept the same.

All functions that were changed kept most of the documentation although expanded in some cases. (Fell free to revert this). Main cases being the `HttpRouter` class and `HttpServer.enable_cors()`

Please at least check my documentation for `HttpServer.enable_cors()` as there wasn't any documentation for it and i'm not 100% sure on how it works. (Although it just felt wrong to have no documentation so i included something)

### CODE REVIEW

File: addons/godottpd/http_request.gd [hunk 1]
```diff
@@ -1,53 +1,53 @@
...
-		
+
 		return data
-	
+
 	# Not supported contenty type parsing... for now
 	return null
 
-# Override `str()` method, automatically called in `print()` function
+# #Override `str()` method, automatically called in `print()` function
```
Comment (dploeger) [inline addons/godottpd/http_request.gd]:
```suggestion
## Override `str()` method, automatically called in `print()` function
```

File: addons/godottpd/http_server.gd [hunk 2]
```diff
@@ -1,19 +1,28 @@
-# A routable HTTP server for Godot
-extends Node
-class_name HttpServer
+## A routable HTTP server for Godot
+##
+## Controlls a back-end localhost server.
```
Comment (dploeger) [inline addons/godottpd/http_server.gd]: I'd rather go with

```suggestion
## Provides a web server with routes for specific endpoints
```

File: addons/godottpd/http_server.gd [hunk 3]
```diff
@@ -257,17 +265,22 @@ func _path_to_regexp(path: String, should_match_subfolders: bool = false) -> Arr
...
 		else:
 			regexp += "/" + fragment
-	regexp += "[/#?]?$" if not should_match_subfolders else "(?<subpath>$|/.*)" 
+	regexp += "[/#?]?$" if not should_match_subfolders else "(?<subpath>$|/.*)"
 	return [regexp, params]
 
 
+## Enable CORS (Cross-origin resource sharing) which only allows requests from the specified servers
+## [br]
+## [br][param allowed_origins] - The origins that area allowed to be access from this sesrver
```
Comment (dploeger) [inline addons/godottpd/http_server.gd]:
```suggestion
## [br][param allowed_origins] - The origins that are allowed to be accessed from this server
```

File: addons/godottpd/http_server.gd [hunk 4]
```diff
@@ -257,17 +265,22 @@ func _path_to_regexp(path: String, should_match_subfolders: bool = false) -> Arr
...
+## [br][param access_control_allowed_methods] - The methods that are allowed to be sent
```
Comment (dploeger) [inline addons/godottpd/http_server.gd]:
```suggestion
## [br][param access_control_allowed_methods] - The methods that are allowed to be used
```

**Review by dploeger — CHANGES_REQUESTED** [review]
This is really great work! Thanks a lot.

You got the CORS stuff right and I know it's hard. I had a hard time studying it as well. 😄

Just a few remarks and we're set I guess. And could I also ask you to change the commit messages? I'm using [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) here. So your commits should start with something like "feat:" or "fix:".

Thanks a lot for your work!

**Comment by dragmine149** [comment]
Updated the title (Please don't mind the multiple attempts) and approved of the requests.

**Comment by dploeger** [comment]
Thanks but I rather talked about changing the commit messages. You can do that using an interactive rebase and reword the commits.

If you don't know how, though, don't worry. I can also simply squash the PR into a single commit that matches the conventional commit message.

**Comment by dragmine149** [comment]
Took me a while to figure out how the interactive rebase worked but got it sorted, and as such updated the commit messages.

**Review by dploeger — APPROVED** [review]
Awesome! Thanks a lot. 🎁

**Comment by dploeger** [comment]
I think I'll release a new version tomorrow as I'm currently only on mobile.

**Pull request merged**